pkg/apis/serving/v1/revision_helpers.go

```diff
@@ -144,9 +143,3 @@ func (rs *RevisionStatus) IsActivationRequired() bool {
 	c := revisionCondSet.Manage(rs).GetCondition(RevisionConditionActive)
 	return c != nil && c.Status != corev1.ConditionTrue
 }
-
-// IsReplicaSetFailure returns true if the deployment replicaset failed to create
-func (rs *RevisionStatus) IsReplicaSetFailure(deploymentStatus *appsv1.DeploymentStatus) bool {
```

Comment:
Is there a way to add a case for https://github.com/knative/serving/issues/14157? I was looking at this function (being removed here) and wondering if we should check if a deployment ever becomes available. In other words when we scale from zero set
```
                    {
                        "status": "False",
                        "type": "ResourcesAvailable"
                    }
```
and when we go back to zero keep that msg. Afaik when we scale down to zero by design we set that condition back to true because we are not active and any availability errors could have been resolved already.  However is not that helpful for reporting status. Wdyth?

Reply:
To clarify further:
```
	// ReplicaFailure is added in a deployment when one of its pods fails to be created
	// or deleted.
```
In the ticket above what is happening is that the pod is being created but availability is never reached but we scale down to zero without persisting the msg that we never managed to get any pods available.

Reply:
> wondering if we should check if a deployment ever becomes available

We have on the PodAutoscaler the condition `ScaleTargetInitialized`. I wonder if it makes sense that we capture autoscaling failures in the 'autoscaling system' and bubble that up via the podautoscaler. 

Thus we keep the revision controller strictly about just reconciling the deployment and not introspecting whether it's failing to scale - seems like there's pros and cons to that.

Thus I think I'd prefer to keep this PR not address #14157 for now